mysensors/mysensors.py

```diff
@@ -280,14 +297,15 @@ def _perform_file_action(self, filename, action):
             raise Exception('Unsupported file type %s' % ext[1:])
         func(filename)
 
-    def alert(self, nid):
+    def alert(self, msg):
         """Tell anyone who wants to know that a sensor was updated.
 
         Also save sensors if persistence is enabled.
         """
         if self.event_callback is not None:
             try:
-                self.event_callback('sensor_update', nid)
+                setattr(msg, 'gateway', self)
```

Comment:
I think adding the gateway as an argument and attribute in the `Message` class is ok and wouldn't be a breaking change. Make it a keyword argument and default to `None`. You'll have to update the copy method as well.

FYI, I've used a "workaround" for home assistant, ie a function factory and passing along the gateway object in the final callback, and it works well.